docs/r2/image_summaries.ipynb

```diff
@@ -0,0 +1,639 @@
+      "cell_type": "code",
+      "source": [
+        "# Download the data. The data is already divided into train and test.\n",
+        "# The labels are integers representing classes.\n",
+        "fashion_mnist = keras.datasets.fashion_mnist\n",
+        "(train_images, train_labels), (test_images, test_labels) = \\\n",
+        "    fashion_mnist.load_data()\n",
+        "\n",
+        "# Names of the integer classes, i.e., 0 -> T-short/top, 1 -> Trouser, etc.\n",
+        "class_names = ['T-shirt/top', 'Trouser', 'Pullover', 'Dress', 'Coat', \n",
+        "    'Sandal', 'Shirt', 'Sneaker', 'Bag', 'Ankle boot']"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "qNsjMY0364j4",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Visualizing a single image\n",
+        "\n",
+        "To understand how the Image Summary API works, you're now going to simply log the first training image in your training set in TensorBoard.\n",
+        "\n",
+        "First, examine the shape of your training data."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "FxMPcdmvBn9t",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "print(\"Shape: \", train_images[0].shape)\n",
+        "print(\"Label: \", train_labels[0], \"->\", class_names[train_labels[0]])"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "4F8zbUKfBuUt",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Notice that the shape of each image in the data set is a rank-2 tensor of shape (28, 28), representing the height and the width.\n",
+        "\n",
+        "However, ```tf.summary.image()``` expects a rank-4 tensor containing ```(batch_size, height, width, channels)```. Therefore, the tensors need to be reshaped. \n",
+        "\n",
+        "You're logging only one image, so ```batch_size``` is 1. The images are grayscale, so set ```channels``` to 1."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "5yPh-7EWB8IK",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Reshape the image for the Summary API.\n",
+        "img = np.reshape(train_images[0], (-1, 28, 28, 1))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "JAdJDY3FCCwt",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "You're now ready to log this image and view it in TensorBoard."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "IJNpyVyxbVtT",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Clear out any prior log data.\n",
+        "!rm -rf logs\n",
```

Comment:
Might be worth separating the cleaning so that multiple images can be logged easily.

Reply:
I generally agree, but in this case it's clearer to the user if the logs are automatically cleared up front.